backend/infrahub/core/schema_manager.py

```diff
@@ -634,6 +634,17 @@ def validate_uniqueness_constraints(self) -> None:
                         allowed_path_types=SchemaElementPathType.ATTR | SchemaElementPathType.REL_ONE_NO_ATTR,
                         element_name="uniqueness_constraints",
                     )
+
+                    if schema_path.is_type_attribute:
+                        required_properties = tuple(
+                            schema_path.attribute_schema.get_class().get_allowed_property_in_path()
+                        )
+                        if schema_path.attribute_property_name not in required_properties:
+                            raise ValueError(
+                                f"{node_schema.kind} uniqueness contraint is invalid at attribute '{schema_path.attribute_schema.name}', it must "
+                                f"end with a property such as: {', '.join(required_properties)}"
```

Comment:
I think `it must end with one of the following properties: ...` make it more clear that it has to be one of the allowed properties